src/java/com/puppetlabs/jruby_utils/pool/JRubyPool.java

```diff
@@ -491,8 +491,7 @@ private void signalIfLockCanProceed() {
         // is active at a time - a caller of lock() that has just acquired
         // the pool lock but is waiting for all registered elements to be
         // returned to the queue.
-        if (registeredElements.size() == liveQueue.size() ||
-                pill != null) {
+        if (this.maxSize == liveQueue.size() || pill != null) {
```

Comment:
This change scares me a bit, I think because I haven't quite been able to think through why we were using `registeredElements` before rather than `maxSize` and when we register/unregister instances and ahhh my brain.

Um, anyway, other than me being just possibly irrationally nervous about this, I think the comment is now out of date - we're not waiting for all registered elements to be returned to the queue, we're waiting for the queue to be the right size.